core/blockchain.go

```diff
@@ -56,7 +56,7 @@ const (
 	blockCacheLimit     = 256
 	maxFutureBlocks     = 256
 	maxTimeFutureBlocks = 30
-	// must be bumped when consensus algorithm is changed, this forces the upgradedb
+	// BlockChainVersion must be bumped when consensus algorithm is changed, this forces the upgradedb
```

Comment:
This isn't really a doc for `BlockChainVersion`, rather just a memo. Alas it's wrong and `upgradedb` is gone a long time ago, so we should rewrite to:

`BlockChainVersion ensures that an incompatible database forces a resync from scratch.`